basic-javascript/project-1.spec.js

```diff
@@ -1,3 +1,347 @@
+    const num3 = roundUp(-2.0222);
+    const num4 = roundUp(true);
+    const num5 = roundUp('sure');
+    
+    expect(num).toBe(8);
+    expect(num2).toBe(98);
+    expect(num3).toBe(-2);
+    expect(num4).toBeNaN;
+    expect(num5).toBeNaN;
+    
+  });
+
+  it('adds an exclamation point', () => {
+    const str = addExclamationPoint('hello');
+    const str2 = addExclamationPoint(1);
+    const str3 = addExclamationPoint(true);
+    
+    expect(str).toBe('hello!');
+    expect(str2).toBe('1!');
+    expect(str3).toBe('true!');
+  });
+
+  it('combines first and last names', () => {
+    const str = combineNames('Pablo', 'Escobar');
+    const str2 = combineNames('fresh', 'hell');
+    const str3 = combineNames(true, false);
+    const str4 = combineNames(1,6);
+
+    expect(str).toBe('Pablo Escobar');
+    expect(str2).toBe('fresh hell');
+    expect(str3).toBe('true false');
+    expect(str4).toBe('1 6');
+  });
+
+  it('greets user', () => {
+    const str = getGreeting('Pablo');
+    const str2 = getGreeting(false);
+    const str3 = getGreeting(86);
+
+    expect(str).toBe('Hello Pablo!');
+    expect(str2).toBe('Hello false!');
+    expect(str3).toBe('Hello 86!');
+  });
+
+  it('calculates area of rectangle', () => {
+    const num = getRectangleArea(7,3);
+    const num2 = getRectangleArea(5,-2);
+    const num3 = getRectangleArea('hello','kelly');
+    const num4 = getRectangleArea(false,true);
+
+    expect(num).toBe(21);
+    expect(num2).toBe(-10); //the original function really should be changed so that you cannot enter negative numbers
+    expect(num3).toBeNaN;
+    expect(num4).toBeNaN;
+  });
+
+  it('calculates area of triangle', () => {
+    const num = getTriangleArea(3,5);
+    const num2 = getTriangleArea(-4,8);
+    const num3 = getTriangleArea('square','circle');
+    const num4 = getTriangleArea(true,true);
+
+    expect(num).toBe(7.5);
+    expect(num2).toBe(-16); //as above, this result makes zero sense in the real world
+    expect(num3).toBeNaN;
+    expect(num4).toBeNaN;
+  });
+
+  it('calculates area of circle', () => {
+    const num = getCircleArea(3);
+    const num2 = getCircleArea(0);
+    const num3 = getCircleArea(-1);
+    const num4 = getCircleArea('what');
+    const num5 = getCircleArea(false);
+
+    expect(num).toBe(28.274333882308138);
+    expect(num2).toBe(0);
+    expect(num3).toBe(3.141592653589793); //this gives an ok result bc we square the radius but it still shouldn't accept a negative number
+    expect(num4).toBeNaN;
+    expect(num5).toBeNaN;
+  });
+
+  it('calculates rectangle prism volume', () => {
+    const num = getRectangularPrismVolume(3,4,5);
+    const num2 = getRectangularPrismVolume(-1,-65,.000394);
+    const num3 = getRectangularPrismVolume('w','t','f');
+    const num4 = getRectangularPrismVolume(true,true,true);
+
+    expect(num).toBe(60);
+    expect(num2).toBe(.025609999999999997);
+    expect(num3).toBeNaN;
+    expect(num4).toBeNaN;
+  });
+  
+});
```

Comment:
Completely agree that some of these tests should be covered from within the function, good work :)